File: InMotionGIT.Common - Copy/Helpers/QuickData.vb
Imports System.Configuration
Imports System.Globalization

Namespace Helpers

    Public Class QuickData

        'Public Shared Function ConvertQuotes(value As String) As String
        '    Return value.Replace("'", "''")
        'End Function

        Public Shared Function QueryExecute(query As String, dbConnection As System.Data.Common.DbConnection) As DataTable
            Using DataAccess As New Data()
                Dim result As DataTable
                With DataAccess
                    result = DataAccess.QueryExecute(query, dbConnection)
                End With
                Return result
            End Using
        End Function

        Public Shared Function QueryExecute(ByVal query As String, ByVal connectionStringName As String) As DataTable
            Using DataAccess As New Data()
                Dim result As DataTable
                With DataAccess
                    result = DataAccess.QueryExecute(query, connectionStringName)
                End With
                Return result
            End Using
        End Function

        Public Shared Function QueryScalar(Of T)(ByVal query As String, ByVal connectionStringName As String) As T
            Using DataAccess As New Data()
                Dim result As T
                With DataAccess
                    result = DataAccess.QueryScalar(Of T)(query, connectionStringName)
                End With
                Return result
            End Using
        End Function

        Public Shared Function CommandExecute(ByVal command As String, ByVal connectionStringName As String) As Long
            Using DataAccess As New Data()
                Dim result As Long
                With DataAccess
                    result = DataAccess.CommandExecute(command, connectionStringName)
                End With
                Return result
            End Using
        End Function

        'TODO: validar este caso con el provider de microsoft.
        Public Shared Function DbProviderParameterPrefix(parameterName As String, providerName As String) As String
            Dim result As String = String.Empty
            Select Case providerName.ToLower
                Case "oracle.dataaccess.client"
                    result = String.Format(":{0}", parameterName)
                Case Else
                    result = String.Format("@{0}", parameterName)
            End Select

            Return result
        End Function

        Public Shared Function DateFormat(ByVal repositoryName As String) As String
            Return ConfigurationManager.AppSettings(String.Format("{0}.DateFormat", repositoryName))
        End Function

        Public Shared Function ValueDateFormat(ByVal repositoryName As String, value As Date) As String
            Dim result As String = value.ToString(QuickData.DateFormat(repositoryName), New CultureInfo("en-US"))
            'result = result.Replace("/mm/", "/")
            Return result
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/Helpers/QuickData.cs
using System;
using System.Configuration;
using System.Data;
using System.Globalization;

namespace InMotionGIT.Common.Helpers
{

    public class QuickData
    {

        // Public Shared Function ConvertQuotes(value As String) As String
        // Return value.Replace("'", "''")
        // End Function

        public static DataTable QueryExecute(string query, System.Data.Common.DbConnection dbConnection)
        {
            using (var DataAccess = new Data())
            {
                DataTable result;
                result = DataAccess.QueryExecute(query, dbConnection);
                return result;
            }
        }

        public static DataTable QueryExecute(string query, string connectionStringName)
        {
            using (var DataAccess = new Data())
            {
                DataTable result;
                result = DataAccess.QueryExecute(query, connectionStringName);
                return result;
            }
        }

        public static T QueryScalar<T>(string query, string connectionStringName)
        {
            using (var DataAccess = new Data())
            {
                T result;
                result = DataAccess.QueryScalar<T>(query, connectionStringName);
                return result;
            }
        }

        public static long CommandExecute(string command, string connectionStringName)
        {
            using (var DataAccess = new Data())
            {
                long result;
                result = DataAccess.CommandExecute(command, connectionStringName);
                return result;
            }
        }

        // TODO: validar este caso con el provider de microsoft.
        public static string DbProviderParameterPrefix(string parameterName, string providerName)
        {
            string result = string.Empty;
            switch (providerName.ToLower() ?? "")
            {
                case "oracle.dataaccess.client":
                    {
                        result = string.Format(":{0}", parameterName);
                        break;
                    }

                default:
                    {
                        result = string.Format("@{0}", parameterName);
                        break;
                    }
            }

            return result;
        }

        public static string DateFormat(string repositoryName)
        {
            return ConfigurationManager.AppSettings[string.Format("{0}.DateFormat", repositoryName)];
        }

        public static string ValueDateFormat(string repositoryName, DateTime value)
        {
            string result = value.ToString(DateFormat(repositoryName), new CultureInfo("en-US"));
            // result = result.Replace("/mm/", "/")
            return result;
        }

    }

}